classes/core/PKPQueueProvider.php

```diff
@@ -139,11 +140,21 @@ public function register()
 
         $this->registerDatabaseConnector(app()->get(\Illuminate\Queue\QueueManager::class));
 
-        if (!Application::get()->isUnderMaintenance() && Config::getVar('queues', 'job_runner', false)) {
+        if (!Application::get()->isUnderMaintenance() && Config::getVar('queues', 'job_runner', true)) {
+            
+            // This is required to make the upgrade test pass when updaring from 3.2.1 and 3.2.0
+            // As for those version we did not have any queue related functionality 
+            // and jobs related tables are missing
+            if (!Schema::hasTable('jobs')) {
```

Comment:
The job runner should not be running during maintenance due to the `isUnderMaintenance` check above; can you look into why that's not catching the upgrade?

Reply:
The implementation was bit misleading . we need to check the maintenance mode with the `register_shutdown_function` as service provider `register` method get called at the initial app bootstrapping time . Also moved the job runner code to `boot` method as `register` is not the right place for such implementation .

Reply:
Thanks, @touhidurabir; is this ready for review/merging?

Reply:
@asmecher Yes, it is . `OMP` test has passed but `OJS/OPS` still running. But should be ready to merge .